### PR TITLE
chore: deprecate RecommendationClient in favor of PersonalizationClient

### PR DESCRIPTION
**Pull request by DevinCodes**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no    
| BC breaks?        | no     
| Need Doc update   | yes

## Describe your change
As per our internal specification, this renames the `RecommendationClient` to `PersonalizationClient`. For backwards compatibility you'll still be able to use the `RecommendationClient` until the next major version.

### CODE REVIEW

File: lib/algolia/personalization_client.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      # @return self
+      #
+      def self.create(app_id, api_key)
+        config = Personalization::Config.new(application_id: app_id, api_key: api_key)
+        create_with_config(config)
+      end
+
+      # Create a new client providing only an Recommendation::Config object
+      #
+      # @param config [Recommendation::Config]
```
Comment (sbellone) [inline lib/algolia/personalization_client.rb:33]:
```suggestion
      # Create a new client providing only an Personalization::Config object
      #
      # @param config [Personalization::Config]
```

File: lib/algolia/config/recommendation_config.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 module Algolia
   module Recommendation
+    # <b>DEPRECATED:</b> Please use <tt>Algolia::Personalization::Config</tt> instead.
     class Config < BaseConfig
```
Comment (julienbourdeau) [inline lib/algolia/config/recommendation_config.rb:4]: Is there a reason you didn't go for the same solution as the client? extend the Personalization version and remove all content in this class?
```suggestion
    class Config < Algolia::Personalization::Config
```

**Review by julienbourdeau — APPROVED** [review]

**Review by sbellone — COMMENTED** [review]
Just need to rename the called endpoint, otherwise looks good 👌

**Comment by DevinCodes** [comment]
Thank you both! I've updated the PR to address your suggestions 🙂

**Review by sbellone — APPROVED** [review]

**Review by sbellone — APPROVED** [review]

**Pull request merged**